### PR TITLE
ISSUE-35:

### PR DESCRIPTION
**Pull request by rouazana**
* GET to Refetch URL endpoints
* Check Authentication header by servlet filter

### CODE REVIEW

**Comment by rouazana** [comment]
test this please

**Comment by aduprat** [comment]
:+1

**Comment by rouazana** [comment]
test this please

**Comment by rouazana** [comment]
test this please

File: protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest httpRequest = (HttpServletRequest)request;
+        String authHeader = httpRequest.getHeader("Authorization");
+        if (authHeader == null || !checkAuthorizationHeader(authHeader)) {
```
Comment (mbaechler) [inline protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationFilter.java]: why the null check is not part for checkAuthorizationHeader ?

File: protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            ((HttpServletResponse)response).sendError(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+        chain.doFilter(request, response);
+    }
+
+    private boolean checkAuthorizationHeader(String authHeader) throws IOException {
```
Comment (mbaechler) [inline protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationFilter.java]: Maybe Optional<String> would handle the null case

File: protocols/jmap/src/main/java/org/apache/james/jmap/BypassOnPostFilter.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest httpRequest = (HttpServletRequest)request;
+        if ("POST".equals(httpRequest.getMethod())) {
+            chain.doFilter(request, response);
+        }
+        nestedFilter.doFilter(httpRequest, response, chain);
```
Comment (mbaechler) [inline protocols/jmap/src/main/java/org/apache/james/jmap/BypassOnPostFilter.java]: it will be called in both cases, is it expected ?

Comment (mbaechler) [inline protocols/jmap/src/main/java/org/apache/james/jmap/BypassOnPostFilter.java]: maybe some unit tests on this class would be a good thing

File: protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationFilter.java (hunk 1 above)
Comment (rouazana) [inline protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationFilter.java]: ok

File: protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationFilter.java (hunk 2 above)
Comment (rouazana) [inline protocols/jmap/src/main/java/org/apache/james/jmap/AuthenticationFilter.java]: ok

File: protocols/jmap/src/main/java/org/apache/james/jmap/BypassOnPostFilter.java (hunk 3 above)
Comment (rouazana) [inline protocols/jmap/src/main/java/org/apache/james/jmap/BypassOnPostFilter.java]: ok, I added some for AuthenticationFilter too.

**Comment by mbaechler** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Comment by rouazana** [comment]
Merged on linagora/jmap (james-project repository)

**Pull request closed without merging**